doc/api/module.md

```diff
@@ -1048,6 +1048,51 @@ returned object contains the following keys:
 * columnNumber: {number} The 1-indexed columnNumber of the
   corresponding call site in the original source
 
+## TypeScript support
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+It is possible to execute TypeScript files by setting the experimental
+flag [`--experimental-strip-types`][].
+Node.js will transpile TypeScript source code into JavaScript source code.
+During the transpilation process, no type checking is performed, and types are discarded.
+
+> The flag [`--experimental-require-module`][] is currently not supported.
+
+### Limitations
+
+#### .mts and .cts
+
+Currently `.mts` and `.cts` extensions are partially supported.
+Due to this feature being very early in development, it is not recommended to mix `.cts` and
+`.mts`.
+
+#### TypeScript only features
+
+By design, Node.js will not perform transformations on TypeScript-only features.
+This means `Enum`, `experimentalDecorators`, `namespaces`, and other TS-only
+features are not supported.
+
+Importing [modules without extension][] is not supported.
+
+```ts
+  import { foo } from 'foo'; // it will not work
+  import { foo } from './foo.ts'; // it will work
+```
+
+#### REPL
+
+Executing source code in the REPL with TypeScript is not supported.
```

Comment:
But `--eval` _is_ supported? Presumably it follows the same rules as JavaScript, where it must be CommonJS syntax unless `--input-type=module` is passed? But I thought CommonJS syntax in TypeScript files isn’t supported yet?

Reply:
Eval it's supported